graphql/resolvers/exerciseSubmissionCrud.ts

```diff
@@ -0,0 +1,46 @@
+import { MutationAddExerciseSubmissionArgs } from '..'
+import { Context } from '../../@types/helpers'
+import prisma from '../../prisma'
+
+export const exerciseSubmissions = (
+  _parent: void,
+  _args: void,
+  context: Context
+) => {
+  const userId = context.req.user?.id
+  if (!userId) return []
+
+  return prisma.exerciseSubmission.findMany({
+    include: {
+      exercise: { include: { module: { include: { lesson: true } } } },
+      user: true
+    },
+    where: { user: { id: userId } }
+  })
+}
+
+export const addExerciseSubmission = async (
+  _parent: void,
+  { exerciseId, userAnswer }: MutationAddExerciseSubmissionArgs,
+  context: Context
+) => {
+  const userId = context.req.user?.id
+  if (!userId) throw new Error('User should be logged in.')
+
+  const exerciseSubmission = await prisma.exerciseSubmission.findFirst({
+    where: { userId, exerciseId }
+  })
+
+  if (exerciseSubmission) {
+    return prisma.exerciseSubmission.update({
+      data: { exerciseId, userId, userAnswer },
+      where: { id: exerciseSubmission.id },
+      include: { exercise: true, user: true }
+    })
+  }
+
+  return prisma.exerciseSubmission.create({
+    data: { exerciseId, userId, userAnswer },
+    include: { exercise: true, user: true }
+  })
+}
```

Comment:
I was thinking that DOJO exercise submissions to exercises should be one-to-one. So if you add an exercise submission and then you update it later, it will just be updated rather than creating a new exercise submission.

To enforce this constraint, I was thinking of making `addExerciseSubmission` update an exercise submission if it already exists rather than having a separate `addExerciseSubmission` and `updateExerciseSubmission`.

Another possible way of enforcing this constraint would be to have separate `addExerciseSubmission` and `updateExerciseSubmission` mutations and throw an error inside `addExerciseSubmission` if an exercise containing the userId and exerciseId already exists.